packages/internal/src/types.d.ts

```diff
@@ -3,6 +3,18 @@ import type { ERef, RemotableBrand } from '@endo/eventual-send';
 import type { Primitive } from '@endo/pass-style';
 import type { Callable } from './utils.js';
 
+/**
+ * A map corresponding with a total function such that `get(key)` is assumed to
+ * always succeed.
+ */
+export type TotalMap<K, V> = {
+  [k in Exclude<keyof Map<K, V>, 'get'>]: Map<K, V>[k];
+} & {
+  get: (key: K) => V;
+};
```

Comment:
handy! I think this would be more clear:
```suggestion
export type TotalMap<K, V> = Omit<Map<K, V>, 'get'> & {
  get: (key: K) => V;
};
```